File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/OutgoingEventInfo.cs
using System;
using Volo.Abp.Data;

namespace Volo.Abp.EventBus.Distributed;

public class OutgoingEventInfo : IHasExtraProperties
{
    public static int MaxEventNameLength;// { get; set; } = 256;

    public ExtraPropertyDictionary ExtraProperties { get; protected set; }

    public ID Id;//  { get; }

    public String EventName;//  { get; }

    public byte[] EventData;//  { get; }

    public Date CreationTime;//  { get; }

    protected OutgoingEventInfo()
    {
        ExtraProperties = new ExtraPropertyDictionary();
        this.SetDefaultsForExtraProperties();
    }

    public OutgoingEventInfo(
        ID id,
        String eventName,
        byte[] eventData,
        Date creationTime)
    {
        Id = id;
        EventName = Check.NotNullOrWhiteSpace(eventName, nameof(eventName), MaxEventNameLength);
        EventData = eventData;
        CreationTime = creationTime;
        ExtraProperties = new ExtraPropertyDictionary();
        this.SetDefaultsForExtraProperties();
    }
}

File: ueueo-frameworks/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/OutgoingEventInfo.java
package com.ueueo.eventbus.distributed;

import com.ueueo.ID;
import com.ueueo.data.objectextending.ExtraPropertyDictionary;
import com.ueueo.data.objectextending.IHasExtraProperties;
import org.apache.commons.lang3.StringUtils;
import org.springframework.util.Assert;

import java.util.Date;

public class OutgoingEventInfo implements IHasExtraProperties {
    public static int MaxEventNameLength = 256;

    private ID id;

    private String eventName;

    private byte[] eventData;

    private Date creationTime;

    private ExtraPropertyDictionary extraProperties;

    protected OutgoingEventInfo() {
        extraProperties = new ExtraPropertyDictionary();
        IHasExtraProperties.Extensions.setDefaultsForExtraProperties(this, null);
    }

    public OutgoingEventInfo(ID id, String eventName, byte[] eventData, Date creationTime) {
        this.id = id;
        Assert.isTrue(StringUtils.isNotBlank(eventName), "EventName must not empty!");
        Assert.isTrue(eventName.length() <= MaxEventNameLength, "EventName must <= " + MaxEventNameLength);
        this.eventName = eventName;
        this.eventData = eventData;
        this.creationTime = creationTime;
        this.extraProperties = new ExtraPropertyDictionary();
        IHasExtraProperties.Extensions.setDefaultsForExtraProperties(this, null);
    }

    @Override
    public ExtraPropertyDictionary getExtraProperties() {
        return extraProperties;
    }

    protected void setExtraProperties(ExtraPropertyDictionary extraProperties) {
        this.extraProperties = extraProperties;
    }

    public ID getId() {
        return id;
    }

    public String getEventName() {
        return eventName;
    }

    public byte[] getEventData() {
        return eventData;
    }

    public Date getCreationTime() {
        return creationTime;
    }
}
